processor/src/host/advice/mod.rs

```diff
@@ -31,22 +36,32 @@ pub use source::AdviceSource;
 /// 3. Merkle store, which contains structured data reducible to Merkle paths. The VM can request
 ///    Merkle paths from the store, as well as mutate it by updating or merging nodes contained in
 ///    the store.
-pub trait AdviceProvider: Sized {
-    // REQUIRED METHODS
-    // --------------------------------------------------------------------------------------------
+///
+/// Advice data is store in-memory using [BTreeMap]s as its backing storage.
+#[derive(Debug, Clone, Default)]
+pub struct AdviceProvider {
+    pub stack: Vec<Felt>,
+    pub map: SimpleAdviceMap,
+    pub store: MerkleStore<SimpleMerkleMap>,
+}
```

Comment:
Instead of `SimpleAdviceMap` could we not use the `AdviceMap` struct from `miden-core`?

Also, now we could greatly simplify the `MerkleStore` implementation in `miden-crypto`. Let's create an issue for this there.